sunpy/net/dataretriever/sources/kanzelhohe.py

```diff
@@ -59,27 +60,37 @@ def _get_url_for_timerange(self, timerange, **kwargs):
         """
         returns list of urls corresponding to given TimeRange.
         """
-        wave = int(kwargs['wavelength'].min.value)
+        wave_float = kwargs['wavelength'].min.value
         table = {6563:['halpha2k/recent', 'halph_fr'], 32768:['caiia', 'caiik_fi'], 5460:['phokada', 'bband_fi']}
-        if (wave == 6563):
-            START_DATE = datetime.datetime(2000, 7, 20, 7, 45, 46)
-        elif (wave == 5460):
-            START_DATE = datetime.datetime(2011, 1, 7, 10, 7, 33)
-        elif (wave == 32768):
-            START_DATE = datetime.datetime(2010, 7, 31, 8, 10, 59)
-        if timerange.start < START_DATE:
-            raise ValueError('Earliest date for which Kanzelhohe data is available is {:%Y-%m-%d}'.format(START_DATE))
-        prefix = "http://cesar.kso.ac.at/{datatype}/"
-        if (wave == 6563):
-            suffix = "%Y/kanz_{datatype1}_%Y%m%d_%H%M%S.fts.gz"
-        else:
-            suffix = "%Y/%Y%m%d/processed/kanz_{datatype1}_%Y%m%d_%H%M%S.fts.gz"
-        url_pattern = prefix + suffix
-        crawler = Scraper(url_pattern, datatype = table[wave][0], datatype1 = table[wave][1])
-        if not timerange:
-            return []
-        result = crawler.filelist(timerange)
-        return result
+        #Checking if value is close enough to a wavelength value.
+        #Converting from one unit to other introduces precision errors.
+        try:
+            da = []
+            da.append(wave_float)
+            for wave_nums in table.keys():
+                db = []
+                db.append(wave_nums)
+                if np.isclose(da, db, 1e-10, 1e-10):
+                    wave = wave_nums
+
+            date_table = {6563: datetime.datetime(2000, 7, 20, 7, 45, 46), 5460: datetime.datetime(2011, 1, 7, 10, 7, 33),
+                          32768: datetime.datetime(2010, 7, 31, 8, 10, 59)}
+            START_DATE = date_table[wave]
+            if timerange.start < START_DATE:
+                raise ValueError('Earliest date for which Kanzelhohe data is available is {:%Y-%m-%d}'.format(START_DATE))
+            prefix = "http://cesar.kso.ac.at/{datatype}/"
+            if (wave == 6563):
+                suffix = "%Y/kanz_{datatype1}_%Y%m%d_%H%M%S.fts.gz"
+            else:
+                suffix = "%Y/%Y%m%d/processed/kanz_{datatype1}_%Y%m%d_%H%M%S.fts.gz"
+            url_pattern = prefix + suffix
+            crawler = Scraper(url_pattern, datatype = table[wave][0], datatype1 = table[wave][1])
+            if not timerange:
+                return []
+            result = crawler.filelist(timerange)
+            return result
+        except:
+            print "Enter wavelength with proper values and units"
```

Comment:
2 things:
 - that `print` is python2, you should use `print()`
 - you should not `print` anywhere, either raise an error or do something, but no printing.